plugins/network-elements/nuage-vsp/src/com/cloud/network/manager/NuageVspManagerImpl.java

```diff
@@ -247,6 +467,26 @@ public boolean deleteNuageVspDevice(DeleteNuageVspDeviceCmd cmd) {
             }
         }
 
+        ConfigurationVO cmsIdConfig = _configDao.findByName("nuagevsp.cms.id");
+        HostVO host = findNuageVspHost(nuageVspDevice.getHostId());
+        if (!auditDomainsOnVsp(host, false, true)) {
+            return false;
+        }
+
+        String nuageVspCmsId = findNuageVspCmsIdForDevice(nuageVspDevice.getId(), cmsIdConfig);
+        SyncNuageVspCmsIdCommand syncCmd = new SyncNuageVspCmsIdCommand(SyncType.UNREGISTER, nuageVspCmsId);
+        SyncNuageVspCmsIdAnswer answer = (SyncNuageVspCmsIdAnswer) _agentMgr.easySend(nuageVspDevice.getHostId(), syncCmd);
+        if (answer != null && answer.getSuccess()) {
+            String currentValue = cmsIdConfig.getValue();
+            String newValue = currentValue.replace(nuageVspDevice.getId() + ":" + answer.getNuageVspCmsId(), "");
+            if (newValue.startsWith(";")) {
```

Comment:
Check that ``newValue`` with ``Strings.isNullOrEmpty()`` to avoid a potential ``NullPointerException``.